src/CommandingOfficers/CommanderStrong.java

```diff
@@ -1,40 +1,163 @@
+    UnitProductionModifier upm = new UnitProductionModifier(TerrainType.AIRPORT, infModel);
+    upm.addProductionPair(TerrainType.SEAPORT, infModel);
+    upm.addProductionPair(TerrainType.HEADQUARTERS, infModel);
+    upm.apply(this); // Passive ability, so don't add it to the COModifier list; just apply it and forget it.
+
+    // Set Strong up with a base damage buff and long-range APCs. These COModifiers are
+    // not added to the modifiers collection so they will not be reverted.
     COModifier strongMod = new CODamageModifier(20); // Give us a nice base power boost.
     strongMod.apply(this);
 
     COMovementModifier moveMod = new COMovementModifier();
     moveMod.addApplicableUnitType(UnitModel.UnitEnum.APC);
     moveMod.apply(this);
+
+    addCommanderAbility(new StrongArmAbility(this));
+    addCommanderAbility(new MobilizeAbility(this));
   }
 
-  public void doAbilityMinor()
+  /**
+   * StrongArm grants Strong a firepower bonus, additional mobility,
+   * and the ability to build Bazooka units from all production buildings and the HQ.
+   */
+  private static class StrongArmAbility extends CommanderAbility
   {
-    COModifier strongerMod = new CODamageModifier(20); // Another 20% with the minor ability active.
-    strongerMod.apply(this);
-    modifiers.add(strongerMod); // Add to the list so the modifier can be reverted next turn.
+    private static final String STRONGARM_NAME = "Strongarm";
+    private static final int STRONGARM_COST = 10;
+    private static final int STRONGARM_BUFF = 20;
+
+    COModifier damageMod = null;
+
+    StrongArmAbility(Commander commander)
+    {
+      super(commander, STRONGARM_NAME, STRONGARM_COST);
+
+      damageMod = new CODamageModifier(STRONGARM_BUFF);
+    }
+
+    @Override
+    protected void perform(GameInstance game)
+    {
+      myCommander.addCOModifier(damageMod);
+
+      // Make bazookas buildable from all production buildings.
+      UnitModel mechModel = myCommander.getUnitModel(UnitModel.UnitEnum.MECH);
+      UnitProductionModifier upm = new UnitProductionModifier(TerrainType.AIRPORT, mechModel);
+      upm.addProductionPair(TerrainType.SEAPORT, mechModel);
+      upm.addProductionPair(TerrainType.HEADQUARTERS, mechModel);
+      myCommander.addCOModifier(upm);
+
+      // Grant foot-soldiers and transports additional movement power.
+      COMovementModifier moveMod = new COMovementModifier();
+      moveMod.addApplicableUnitType(UnitModel.UnitEnum.APC);
+      moveMod.addApplicableUnitType(UnitModel.UnitEnum.INFANTRY);
+      moveMod.addApplicableUnitType(UnitModel.UnitEnum.MECH);
+      myCommander.addCOModifier(moveMod);
+
+      // Grant transports extra capacity
+      UnitRemodelModifier urm = new UnitRemodelModifier();
+      for( UnitModel.UnitEnum type : highCapacityUnitModels.keySet() )
+      {
+        urm.addUnitRemodel(myCommander.getUnitModel(type), highCapacityUnitModels.get(type));
+      }
+      myCommander.addCOModifier(urm);
```

Comment:
So uh, it's worth noting that swapping in the high capacity unit models actually nerfs their movement range.
The high capacity models do not have even the D2D move boost, and don't benefit from the movement boosts either.

I suppose the simplest solution is to just make separate copies of the remodeled units.

Reply:
hrm. Good catch. I should just make it possible to clone UnitModels.

Reply:
Done.